src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/BinaryLogEventDataReader.java

```diff
@@ -0,0 +1,317 @@
+        } else if (b == 254) {
+            return buffer.getLong();
+        }
+        throw new RuntimeException("Unexpected packed number byte " + b);
+    }
+
+    /**
+     * Reads packed integer value from the current position in the buffer. Similar to {@link #readPackedLong()} but
+     * makes sure that the number read is fit into integer.
+     * @throws IOException if unexpected bytes are encountered or in case of integer overflow (IOException is used for
+     *                      compatibility with {@link com.github.shyiko.mysql.binlog.io.ByteArrayInputStream}.
+     */
+    public int readPackedInteger() throws IOException {
+        long value = readPackedLong();
+        if (value > Integer.MAX_VALUE || value < Integer.MIN_VALUE) {
+            throw new IOException("Packed integer " + value + " is out of Integer range");
+        }
+        return (int)value;
+    }
+
+
+    public BitSet readBitSet(int length, boolean bigEndian) {
+        // According to MySQL internals the amount of storage required for N columns is INT((N+7)/8) bytes.
+        int lengthInBytes = (length + 7) >> 3;
+        byte[] bytes = new byte[lengthInBytes];
+        buffer.get(bytes);
+
+        // BitSet.valueOf we're using below interprets bytes as little-endian
+        if (!bigEndian) {
+            bytes = reverse(bytes);
+        }
+
+        // Zero out not used bits.
+        byte mask = (byte)(0xff >> (bytes.length * 8 - length));
+        bytes[bytes.length - 1] = (byte)(bytes[bytes.length - 1] & mask);
+
+        return BitSet.valueOf(bytes);
+    }
+
+    public int available() {
+        return buffer.remaining();
+    }
+
+    /**
+     * Reads a string of size {@code length} from the current position in the buffer.
+     */
+    public String readString(int length) {
+        byte[] bytes = new byte[length];
+        buffer.get(bytes);
+        return new String(bytes);
+    }
+
+    /**
+     * Reads a string prepended by a packed integer length from the current position in the buffer.
+     */
+    public String readLengthEncodedString() throws IOException {
+        return readString(readPackedInteger());
+    }
+
+    /**
+     * Reads a zero-terminated string from the current position in the buffer.
+     */
+    public String readZeroTerminatedString() throws IOException {
+        int zeroIndex = indexOf((byte)0);
+        if (zeroIndex == -1) {
+            throw new IOException("End of buffer reached with no zero byte found");
+        }
+
+        int length = zeroIndex - buffer.position();
+        String value = readString(length);
+        buffer.position(buffer.position() + 1);
+        return value;
+    }
+
+    /**
+     * Reads a zero-terminated string of length {@code lengthHint} from the current position in the buffer.
+     * This method is usually faster than {@link #readZeroTerminatedString()} because we can avoid searching for a
+     * zero byte.
```

Comment:
This seems like a nice optimization as long as we can rely on db name length and table name length

Reply:
Well, according to documentation (and experiments), we do have length provided for these values.